### ISSUE TITLE
Object.keys() does not guarantee order that lead to unexpected comparaison failure

### ISSUE THREAD

**Issue by ccordenier** [zaggino/z-schema#222]
We have a schema made of complex enum something like :
 
```
{
  "type":"object",
  "oneOf": [
    {
      "type": "object",
      "properties": {
        "code": {
          "type": "string"
        },
        "libelle": {
          "type": "string"
        }
      },
      "enum": [
        {
          "code": "null",
          "libelle": "null"
        }
      ]
    },
    {
      "type": "object",
      "properties": {
        "code": {
          "type": "string"
        },
        "libelle": {
          "type": "string"
        }
      },
      "enum": [
        {
          "code": "0",
          "libelle": "Choice 1"
        }
      ]
    },
    {
      "type": "object",
      "properties": {
        "code": {
          "type": "string"
        },
        "libelle": {
          "type": "string"
        }
      },
      "enum": [
        {
          "code": "1",
          "libelle": "Choice 2"
        }
      ]
    },
    {
      "type": "object",
      "properties": {
        "code": {
          "type": "string"
        },
        "libelle": {
          "type": "string"
        }
      },
      "enum": [
        {
          "code": "2",
          "libelle": "Choice 3"
        }
      ]
    }
  ]
}

```

This fails on validation with z-schema but pass the validation on https://www.jsonschemavalidator.net/

After debugging it happens to be object equality check that leads to validation failure. And it's due to a different of order in object keys set :

cf. https://github.com/zaggino/z-schema/blob/10263d992260e3d34e49881174babf7c83eaa393/src/Utils.js#L87

Using 'sort' on keys before iterating properties should solve this issue without compromising the whole algorithm ?

**Comment by zaggino** [comment]
Fixed in `z-schema@3.24.1`

**Comment by ccordenier** [comment]
Thank you !
